tools/clang/test/CodeGenHLSL/batch/declarations/resources/byteaddressbuffers/load_type_shapes_sm62.hlsl

```diff
@@ -0,0 +1,38 @@
+// RUN: %dxc -E main -T vs_6_2 %s | FileCheck %s
+
+// Tests that ByteAddressBuffer.Load<T> works with all type shapes with SM 6.2
+
+struct S { int i; float f; };
+ByteAddressBuffer buf;
+RWStructuredBuffer<int> out_scalar;
+RWStructuredBuffer<int2> out_vector;
+RWStructuredBuffer<int2x2> out_matrix;
+RWStructuredBuffer<int[2]> out_array;
+RWStructuredBuffer<S> out_struct;
+RWStructuredBuffer<S[2]> out_struct_array;
+
+void main() {
+  // CHECK: call %dx.types.ResRet.i32 @dx.op.rawBufferLoad.i32(i32 139, %dx.types.Handle {{.*}}, i32 100, i32 undef, i8 1, i32 4)
+  out_scalar[0] = buf.Load<int>(100);
+  
+  // CHECK: call %dx.types.ResRet.i32 @dx.op.rawBufferLoad.i32(i32 139, %dx.types.Handle {{.*}}, i32 200, i32 undef, i8 3, i32 4)
+  out_vector[0] = buf.Load<int2>(200);
+  
+  // CHECK: call %dx.types.ResRet.i32 @dx.op.rawBufferLoad.i32(i32 139, %dx.types.Handle {{.*}}, i32 300, i32 undef, i8 15, i32 4)
+  out_matrix[0] = buf.Load<int2x2>(300);
+
+  // CHECK: call %dx.types.ResRet.i32 @dx.op.rawBufferLoad.i32(i32 139, %dx.types.Handle {{.*}}, i32 400, i32 undef, i8 1, i32 4)
+  // CHECK: call %dx.types.ResRet.i32 @dx.op.rawBufferLoad.i32(i32 139, %dx.types.Handle {{.*}}, i32 404, i32 undef, i8 1, i32 4)
+  out_array[0] = buf.Load<int[2]>(400);
+  
+  // CHECK: call %dx.types.ResRet.i32 @dx.op.rawBufferLoad.i32(i32 139, %dx.types.Handle {{.*}}, i32 500, i32 undef, i8 1, i32 4)
+  // CHECK: call %dx.types.ResRet.f32 @dx.op.rawBufferLoad.f32(i32 139, %dx.types.Handle {{.*}}, i32 504, i32 undef, i8 1, i32 4)
+  out_struct[0] = buf.Load<S>(500);
```

Comment:
@MrTrillian More questions! :)
Is Load templated with a struct only allowed if the struct members are numeric types?

e.g. I hope something like this is disallowed?
```hlsl
struct S {
  ByteAddressBuffer b;
  RWByteAddressBuffer c;
};

ByteAddressBuffer a;
// ...
// inside 'main':
S myS = a.Load<S>(0);
float4 myFloat4 = myS.b.Load<float4>(myS);
```

This currently gives me:
```
error: Explicit template arguments on intrinsic Load must be a single numeric type
```

Reply:
Correct, we use `hlsl::IsHLSLNumericOrAggregateOfNumericType`. Semantic analysis should catch this so you wouldn't have to handle it in your backend.

Reply:
\o/ Excellent! Thanks for your quick response